web/src/reducers/activities.js

```diff
@@ -0,0 +1,49 @@
+import { ADD_ACTIVITY, UPDATE_ACTIVITY } from '../actions/activities';
+
+const newActivity = id => ({
+  id,
+  name: '',
+  types: ['HIT'],
+  descShort: '',
+  descLong: '',
+  altApproach: ''
+});
+
+const updateEntry = (state, action) => {
+  const { id, name, value } = action.data;
+  const prior = state.byId[id];
+
+  return {
+    ...state,
+    byId: {
+      ...state.byId,
+      [id]: { ...prior, [name]: value }
+    }
+  };
+};
+
+const initialState = {
+  byId: { 1: newActivity(1) },
+  allIds: [1]
+};
+
+const reducer = (state = initialState, action) => {
+  switch (action.type) {
+    case ADD_ACTIVITY: {
+      const id = Math.max(...state.allIds, 0) + 1;
```

Comment:
Writing this out for my own benefit...  Creating an ID here is safe because when we sync to the API, we can refresh the redux state to match the "real" ID and that will cascade through the app and everything will be okay.  🤔

I'm not sure that's true, but let's fix it when it happens rather than spend any time theorizing about what might happen in the future since it can't possibly happen right now.

Reply:
yep, that was my thought process as well